source/differentiation/sd/symengine_number_types.cc

```diff
@@ -47,27 +47,37 @@ namespace Differentiation
 
     Expression::Expression()
       : expression()
+      , hash(0)
+      , hash_computed(false)
     {}
 
 
     Expression::Expression(const bool value)
       : expression(SE::boolean(value))
+      , hash(0)
+      , hash_computed(false)
```

Comment:
To initialise `hash` and `hash_computed` I could use a delegating constructor here that calls the default constructor. Opinions?